data-prepper-api/src/main/java/org/opensearch/dataprepper/model/sink/AbstractSink.java

```diff
@@ -34,6 +35,11 @@ public AbstractSink(final PluginSetting pluginSetting, int numRetries, int waitT
         retryThread = null;
         this.maxRetries = numRetries;
         this.waitTimeMs = waitTimeMs;
+        sinkContext = (SinkContext)pluginSetting.getAttributeOrDefault("sink_context", null);
+    }
+
+    public SinkContext sinkContext() {
```

Comment:
I don't think we need this method. Each `AbstractSink` has a constructor which is called first. Let the `SinkContext` come in to that constructor.

Reply:
This would end up requiring all sink constructors to change instead of just one. Is that worth not using PluginSetting in AbstractSink?

Reply:
If we really are trying to get away from PluginSetting then I guess it makes sense